source/nordic_sdk/components/ble/common/ble_srv_common.h

```diff
@@ -240,13 +243,65 @@ static __INLINE bool ble_srv_is_indication_enabled(uint8_t * p_encoded_data)
 
-/**@brief Function for making UTF-8 structure refer to an ASCII string.
+/**@brief Function for making a UTF-8 structure refer to an ASCII string.
  *
  * @param[out]  p_utf8   UTF-8 structure to be set.
  * @param[in]   p_ascii  ASCII string to be referred to.
  */
 void ble_srv_ascii_to_utf8(ble_srv_utf8_str_t * p_utf8, char * p_ascii);
 
+
+/**@brief Security Access enumeration.
+ * @details This enumeration gives the possible requirements for accessing a characteristic value.
+ */
+typedef enum
+{
+    SEC_NO_ACCESS    = 0,            /**< Not possible to access. */
+    SEC_OPEN         = 1,            /**< Access open. */
+    SEC_JUST_WORKS   = 2,            /**< Access possible with 'Just Works' security at least. */
+    SEC_MITM         = 3,            /**< Access possible with 'MITM' security at least. */
+    SEC_SIGNED       = 4,            /**< Access possible with 'signed' security at least. */
+    SEC_SIGNED_MITM  = 5             /**< Access possible with 'signed and MITM' security at least. */
+}security_req_t;
+
+
+/**@brief Add characteristic parameters structure.
+ * @details This structure contains the parameters needed to use the @ref characteristic_add function.
+ */
+typedef struct
+{
+    uint16_t                    uuid;                     /**< Characteristic UUID (16 bits UUIDs).*/
+    uint8_t                     uuid_type;                /**< Base UUID. If 0, the Bluetooth SIG UUID will be used. Otherwise, this should be a value returned by @ref sd_ble_uuid_vs_add when adding the base UUID.*/
+    uint16_t                    max_len;                  /**< Maximum length of the characteristic value.*/
+    uint16_t                    init_len;                 /**< Initial length of the characteristic value.*/
+    uint8_t *                   p_init_value;             /**< Initial encoded value of the characteristic.*/
+    bool                        is_var_len;               /**< Indicates if the characteristic value has variable length.*/
+    ble_gatt_char_props_t       char_props;               /**< Characteristic properties.*/
+    bool                        is_defered_read;          /**< Indicate if deferred read operations are supported.*/
+    bool                        is_defered_write;         /**< Indicate if deferred write operations are supported.*/
+    security_req_t              read_access;              /**< Security requirement for reading the characteristic value.*/
+    security_req_t              write_access;             /**< Security requirement for writing the characteristic value.*/
+    security_req_t              cccd_write_access;        /**< Security requirement for writing the characteristic's CCCD.*/
+    bool                        is_value_local;           /**< Indicate if the content of the characteristic is to be stored locally or in the stack.*/
+} ble_add_char_params_t;
+
+
+/**@brief Function for adding a characteristic to a given service. 

```

Comment:
Thanks for tracking this in JIRA

From: LIYOU ZHOU [mailto:notifications@github.com]
Sent: 14 December 2015 13:40
To: ARMmbed/nrf51-sdk
Cc: Rohit Grover
Subject: Re: [nrf51-sdk] Update files to Nordic sdk 9.0.0 (#8)


In source/nordic_sdk/components/ble/common/ble_srv_common.h<https://github.com/ARMmbed/nrf51-sdk/pull/8#discussion_r47497728>:

> +    uint8_t                     uuid_type;                /**< Base UUID. If 0, the Bluetooth SIG UUID will be used. Otherwise, this should be a value returned by @ref sd_ble_uuid_vs_add when adding the base UUID.*/

> +    uint16_t                    max_len;                  /**< Maximum length of the characteristic value.*/

> +    uint16_t                    init_len;                 /**< Initial length of the characteristic value.*/

> +    uint8_t *                   p_init_value;             /**< Initial encoded value of the characteristic.*/

> +    bool                        is_var_len;               /**< Indicates if the characteristic value has variable length.*/

> +    ble_gatt_char_props_t       char_props;               /**< Characteristic properties.*/

> +    bool                        is_defered_read;          /**< Indicate if deferred read operations are supported.*/

> +    bool                        is_defered_write;         /**< Indicate if deferred write operations are supported.*/

> +    security_req_t              read_access;              /**< Security requirement for reading the characteristic value.*/

> +    security_req_t              write_access;             /**< Security requirement for writing the characteristic value.*/

> +    security_req_t              cccd_write_access;        /**< Security requirement for writing the characteristic's CCCD.*/

> +    bool                        is_value_local;           /**< Indicate if the content of the characteristic is to be stored locally or in the stack.*/

> +} ble_add_char_params_t;

> +

> +

> +/**@brief Function for adding a characteristic to a given service.

@rgrover<https://github.com/rgrover> Ticket created http://jira.arm.com/browse/IOTSFW-1414 .

—
Reply to this email directly or view it on GitHub<https://github.com/ARMmbed/nrf51-sdk/pull/8/files#r47497728>.

IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.